library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+// Create a new Envoy instance. The Envoy runner NSThread is started as part of instance
+// initialization with the configuration provided.
+- (id)initWithConfig:(NSString *)config {
+  self.runner = [[NSThread alloc] initWithTarget:self selector:@selector(run:) object:config];
+  [self.runner start];
```

Comment:
What are your thoughts on exposing a `start` function instead of starting the instance here? The reason I ask is that this implementation requires consumers to create an instance of `Envoy` and keep it alive, but they may never need to call into it. In Swift this can lead to unused variable warnings. In practice, consumers would usually call `start()` immediately after instantiating

Reply:
I originally had it set up this way. I backtracked because Java's best run-once guarantees are built into object initialization. So I updated this to match.

Once the direct API is in place, consumers will be making other calls to the library. And on the other hand, someone may have a real reason to want a separate `start` call at some point.

If this ends up being an issue, I don't mind changing it, but for now, it simplified things on the Java side.

Reply:
Okay that's fine with me, we can revisit later if needed

Reply:
Cool, yep can definitely revisit.